modules/e2ee/E2EEncryption.js

```diff
@@ -218,15 +219,25 @@ export class E2EEncryption {
 
     /**
      * Advances the current key by using ratcheting.
-     * TODO: not yet implemented, we are just rotating the key at the moment,
-     * which is a heavier operation.
      *
      * @private
      */
     async _ratchetKeyImpl() {
         logger.debug('Ratchetting key');
 
-        return this._rotateKey();
+        const material = await crypto.subtle.importKey('raw', this._key, 'HKDF', false, [ 'deriveBits' ]);
+        const newKey = await crypto.subtle.deriveBits({
```

Comment:
can we share this between worker and lib as a module?

Reply:
Yes, I was thinking we do this in that "Keys" object we've been talking about.

Reply:
later then :-)